shell/platform/darwin/ios/rendering_api_selection.mm

```diff
@@ -21,26 +21,33 @@ bool ShouldUseMetalRenderer() {
   // past iOS 10.0. The processor was selected as it is the first version at which Metal was
   // supported. The iOS version floor was selected due to the availability of features used by Skia.
   bool ios_version_supports_metal = false;
-  if (@available(iOS 10.0, *)) {
+  if (@available(iOS METAL_IOS_VERSION_BASELINE, *)) {
     auto device = MTLCreateSystemDefaultDevice();
     ios_version_supports_metal = [device supportsFeatureSet:MTLFeatureSet_iOS_GPUFamily1_v3];
   }
   return ios_version_supports_metal;
 }
 #endif  // FLUTTER_SHELL_ENABLE_METAL
 
-IOSRenderingAPI GetRenderingAPIForProcess() {
-#if TARGET_IPHONE_SIMULATOR
-  return IOSRenderingAPI::kSoftware;
-#endif  // TARGET_IPHONE_SIMULATOR
+IOSRenderingAPI GetRenderingAPIForProcess(bool force_software) {
+  if (force_software) {
```

Comment:
This allows software rendering on devices. This check should only be performed on simulators.

Reply:
Done

Reply:
Added an #elif block if we're not in release to print a warning that you can't use this on physical devices.